### PR TITLE
fix issue#2730

### LINKED ISSUE #2730: [ioctl] new accounts being generated unintentionally
### What version of iotex-core are you using (`ioctl version`)?
master

### What operating system and processor architecture are you using?
macOS

### What did you do? If possible, provide a recipe for reproducing the error.
after using ioctl for a while, new accounts are being generated (w/o calling `ioctl account create` to add new account)

### What did you expect to see?
no new account should be generated w/o calling `ioctl account create`

### What did you see instead?
`ioctl account list`, can see anonymous new accounts are added

### PR DESCRIPTION
**Pull request by Liuhaai**
fix #2730

### CODE REVIEW

File: ioctl/newcmd/account/accountdelete_test.go [hunk 1]
```diff
@@ -24,14 +28,21 @@ func TestNewAccountDelete(t *testing.T) {
 	client.EXPECT().SelectTranslation(gomock.Any()).Return("mockTranslationString",
 		config.English).AnyTimes()
 
-	client.EXPECT().GetAddress(gomock.Any()).Return("io1kuz56knh9g7x3ht36t2pgfd8fz5g6rxrh5a8dn", nil)
-
-	ks := keystore.NewKeyStore(config.ReadConfig.Wallet,
+	ks := keystore.NewKeyStore("./",
```
Comment (dustinxie) [inline ioctl/newcmd/account/accountdelete_test.go:31]: let's create the keystore under a temp file folder, like in other tests

Comment (dustinxie) [reply ioctl/newcmd/account/accountdelete_test.go:31]: and delete the temp file, like in other tests

```
defer func() {
		testutil.CleanupPath(t, testPath)
	}()
```

File: ioctl/newcmd/account/accountdelete_test.go [hunk 2]
```diff
@@ -24,14 +28,21 @@ func TestNewAccountDelete(t *testing.T) {
...
 	client.EXPECT().NewKeyStore(gomock.Any(), gomock.Any(), gomock.Any()).Return(ks)
+
 	cmd := NewAccountDelete(client)
-	result, err := util.ExecuteCmd(cmd)
-	require.NotNil(t, result)
-	require.Error(t, err)
+	_, err := util.ExecuteCmd(cmd)
+	require.NoError(t, err)
+
+	files, _ := ioutil.ReadDir(".")
```
Comment (dustinxie) [inline ioctl/newcmd/account/accountdelete_test.go:42]: and no need for this one

**Review by dustinxie — APPROVED** [review]

File: ioctl/newcmd/account/accountdelete_test.go (hunk 1 above)
Comment (Liuhaai) [reply ioctl/newcmd/account/accountdelete_test.go:31]: How can the file name of tmp keystore be acquired in this test?

Comment (dustinxie) [reply ioctl/newcmd/account/accountdelete_test.go:31]:
```
testPath, err := testutil.PathOfTempFile("test-delete")
r.NoError(err)
```
testPath is the filename

File: ioctl/newcmd/account/accountdelete_test.go [hunk 3]
```diff
@@ -24,14 +28,21 @@ func TestNewAccountDelete(t *testing.T) {
 	client.EXPECT().SelectTranslation(gomock.Any()).Return("mockTranslationString",
 		config.English).AnyTimes()
 
-	client.EXPECT().GetAddress(gomock.Any()).Return("io1kuz56knh9g7x3ht36t2pgfd8fz5g6rxrh5a8dn", nil)
-
-	ks := keystore.NewKeyStore(config.ReadConfig.Wallet,
+	ks := keystore.NewKeyStore(os.TempDir(),
```
Comment (dustinxie) [inline ioctl/newcmd/account/accountdelete_test.go:31]: i think better to create a new folder inside the tempDir, and delete this folder after test, something like:
```
testDir = NewDir(os.TempDir())
defer func() {
    os.RemoveDir(testDir)
}
```

File: ioctl/newcmd/account/accountdelete_test.go [hunk 4]
```diff
@@ -24,14 +28,21 @@ func TestNewAccountDelete(t *testing.T) {
...
+
 	cmd := NewAccountDelete(client)
-	result, err := util.ExecuteCmd(cmd)
-	require.NotNil(t, result)
-	require.Error(t, err)
+	_, err := util.ExecuteCmd(cmd)
+	require.NoError(t, err)
+
+	files, _ := ioutil.ReadDir(os.TempDir())
+	for _, file := range files {
```
Comment (dustinxie) [inline ioctl/newcmd/account/accountdelete_test.go:43]: we don't need this for loop

**Pull request closed without merging**